resource/subfunctions/kerberos_realm.md

```diff
@@ -0,0 +1,274 @@
+
+## `ResourceAdmin.add_kerberos_realm()`
+
+```python
+  def add_kerberos_realm( self, kerberos_realm_name: str, traits: dict = {} ) -> Union[dict, bytes]:
+```
+
+#### 📄 Description
+
+&nbsp;
+
+{: .experimental }
+> _Only a subset of available **Segments** and **Traits** are considered **Stable**. See [Segments](../../base/segments_traits_operators#segments) and [Traits](../../base/segments_traits_operators#traits) for more details._
+
+&nbsp;
+
+Define a new **Kerberos Realm** profile in the **REALM** class.
+
+#### 📥 Parameters
+* `kerberos_realm_name`<br>
+  The name of the **Kerberos Realm** profile being defined to the **REALM** class.
+
+* `traits`<br>
+  A dictionary of **traits/attributes** that should be given to the resource on creation. See [Traits](../../base/segments_traits_operators#traits) to see what all of the valid **Resource Traits** are.
+
+#### 📤 Returns
+* `Union[dict, bytes]`<br>
+  Returns a **Security Result dictionary** or **Security Request XML bytes** if the `ResourceAdmin.generate_requests_only` class attribute is set to `True`.
+
+#### ❌ Raises
+* `SecurityRequestError`<br>
+  Raises `SecurityRequestError` when the **Return Code** of a **Security Result** returned by IRRSMO00 is **NOT** equal to `0`.
+* `AddOperationError`<br>
+  Raises `AddOperationError` when the **general resource profile** cannot be added because it already exists.
+* `SegmentTraitError`<br>
+  Raises `SegmentTraitError` when the dictionary of **traits/attributes** provided contains one or more **unknown** traits.
+
+#### 💻 Example
+
+###### Python REPL
+```python
+from pyracf import ResourceAdmin
+resource_admin = ResourceAdmin()
+resource_admin.add_kerberos_realm("TSTREALM")
+{'securityResult': {'resource': {'name': 'TSTREALM', 'class': 'REALM', 'operation': 'set', 'requestId': 'ResourceRequest', 'commands': [{'safReturnCode': 0, 'returnCode': 0, 'reasonCode': 0, 'image': 'RDEFINE REALM             (TSTREALM) ', 'messages': ['ICH10006I RACLISTED PROFILES FOR REALM WILL NOT REFLECT THE ADDITION(S) UNTIL A SETROPTS REFRESH IS ISSUED.']}]}, 'returnCode': 0, 'reasonCode': 0}}
+```
+
+###### Security Result Dictionary as JSON
+```json
+{
+  "securityResult":{
+    "resource":{
+      "name":"TSTREALM",
+      "class":"REALM",
+      "operation":"set",
+      "requestId":"ResourceRequest",
+      "commands":[
+        {
+          "safReturnCode":0,
+          "returnCode":0,
+          "reasonCode":0,
+          "image":"RDEFINE REALM             (TSTREALM) ",
+          "messages":[
+            "ICH10006I RACLISTED PROFILES FOR REALM WILL NOT REFLECT THE ADDITION(S) UNTIL A SETROPTS REFRESH IS ISSUED."
+          ]
+        }
+      ]
+    },
+    "returnCode":0,
+    "reasonCode":0
+  }
+}
+```
+
+## `ResourceAdmin.alter_kerberos_realm()`
+
+```python
+  def alter_kerberos_realm( self, kerberos_realm_name: str, traits: dict = {} ) -> Union[dict, bytes]:

```

Comment:
```python
def alter_kerberos_realm(self, kerberos_realm_name: str, traits: dict = {}) -> Union[dict, bytes]:
```

Reply:
Also remove leading spaces in front of function definition.

Reply:
Leading spaces and unnecessary trailing spaces and indents removed across all function definitions.